### DISCUSSION TITLE
Script Output Options

### DISCUSSION THREAD

**Discussion by axodys** [johnlindquist/kit#462] [Q&A] [answered]
I stumbled upon ScriptKit yesterday while looking into options for writing shell scripts in javascript and it's really cool, but I have a couple questions about how output is handled. My first script attempt uses `exec()` to get a random quote from `fortune` and put it into a text string.  I figured out how to display the resulting text via the `div()` function, but wanted  to display it as ~80 character wide monospaced text. I wrapped the output in `<pre>` tags, but found that it gets cut off by the default size of the ScriptKit window. Is there a way to expand the viewable area? Alternatively how do I configure my shell (I use fish) to execute ScriptKit scripts? I tried the sync path script, but it doesn't seem to do anything for the relevant version of path. Being able to run the same script in terminal or the app itself and have it output in the appropriate place would be awesome, but the terminal side of ScriptKit is a bit opaque to me at the moment.

**Comment by johnlindquist** [comment, marked as answer]
## Resize window

You should be able to drag to resize/position any script window to whatever size you want. Maybe I'm misunderstanding the question though...

![CleanShot 2021-09-29 at 17 11 36](https://user-images.githubusercontent.com/36073/135360845-686c78a7-88f2-43fa-88f5-e23e42857a1c.png)

## Running scripts in the terminal

The terminal command scripts are generated in the `~/.kenv/bin` directory.

So you can either run them like `~/.kenv/bin/some-command` or add `~/.kenv/bin` to your path (here's a helper for it in the app:


![CleanShot 2021-09-29 at 17 12 38](https://user-images.githubusercontent.com/36073/135360922-2d260052-47ed-458f-8dfd-52e42e857a52.png)

you can then just run `some-command` in the terminal.

p.s. - `sync-path` enables the App to run the same tools that your terminal has access to. Like `nvm`, `yarn`, `ruby`, or other tools that might be in custom paths for your terminal.

**Reply by axodys** [reply]
Thanks for clarifying, I completely missed that the ScriptKit window was resizable. I like that the app seems to retain the window size a given script has been adjusted to. Now that I've also got the script running properly in the shell, is there a simple way for it to check which environment it's running in so I can output plaintext in the terminal instead of the html version I pass to the app?

**Reply by johnlindquist** [reply]
The simplest way is to check `env.KIT_CONTEXT`. Kit.app will assign it to "app", otherwise it's undefined.

```js
if (env.KIT_CONTEXT === "app") {
  await div(`This is the app`)
} else {
  console.log(`This is the terminal`)
}
```

**Reply by axodys** [reply]
That is perfect, thank you!

**Reply by johnlindquist** [reply]
I'm very happy to help. Please keep the questions coming. I'm working on docs and I need to know which parts are confusing! 😊
